pkg/controller/ipam/multi_cidr_range_allocator.go

```diff
@@ -263,24 +263,6 @@ func NewMultiCIDRRangeAllocator(
 		logger.Info("No Secondary Service CIDR provided. Skipping filtering out secondary service addresses")
 	}
 
-	if nodeList != nil {
-		for _, node := range nodeList.Items {
-			if len(node.Spec.PodCIDRs) == 0 {
-				logger.V(4).Info("Node has no CIDR, ignoring", "node", klog.KObj(&node))
-				continue
-			}
-			logger.Info("Node has CIDR, occupying it in CIDR map", "node", klog.KObj(&node), "podCIDRs", node.Spec.PodCIDRs)
-			if err := ra.occupyCIDRs(logger, &node); err != nil {
```

Comment:
We can just log the error, and not fail

I think we still need to occupy CIDRs during the bootstrap, as once the controller is up and running and if it processes a node with no available CIDR before it processes the old node, the controller might end up assigning a duplicate CIDR to the new node

Reply:
Let me clarify what I meant by old node, just for the sake of an example, this is especially relevant in the single node controlplane use case:

old node = Node processed by the node-ipam-controller when it was running, and has a podCIDR assigned
new node = Node created while the node-ipam-controller was restarting, has no podCIDRs assigned

Let's say the node-ipam-controller restarts:

Currently, In this loop, we are checking for the nodes with podCIDRs already allocated and marking it as occupied in the in-memory map. 

My concern is, if we don't occupy the used CIDRs during bootstrap and we encounter a new node before we encounter the old node, we might assign the podCIDR assigned to the old node incorrectly to the new node, incorrectly assuming that the CIDR was available and not assigned to any node

> The way the informer works is, it will generate an Add() event for each Node in the informer cache, that IIUIC will have the same effect than looping over all the Nodes before we start.

I might be missing something,  but  I think even in the case of Add() events we cannot rely on the order in which the nodes are added to the informer cache

Reply:
I see, this is clear now and I understand it ... Let me fix it

Reply:
@sarveshr7 PTAL , updated to log only